compiler/packages/babel-plugin-react-compiler/src/Inference/InferMutationAliasingEffects.ts

```diff
@@ -309,8 +311,14 @@ function applySignature(
   ) {
     const aliasingEffects =
       instruction.value.loweredFunc.func.aliasingEffects ?? [];
+    const context = new Set(
+      instruction.value.loweredFunc.func.context.map(p => p.identifier.id),
+    );
     for (const effect of aliasingEffects) {
       if (effect.kind === 'Mutate' || effect.kind === 'MutateTransitive') {
+        if (!context.has(effect.value.identifier.id)) {
+          continue;
+        }
```

Comment:
function expression effects now include effects for params, but these values won't exist in the outer context so we skip them